src/engine/sparqlExpressions/SparqlExpressionValueGetters.h

```diff
@@ -118,6 +120,53 @@ struct StringValueGetter {
 
   std::optional<string> operator()(IdOrString s,
                                    const EvaluationContext* ctx) const {
+    return std::visit(
+        [self = this, ctx](auto el) { return self->operator()(el, ctx); },
+        std::move(s));
+  }
+};
+using StringValueGetter = StringValueGetterImpl<true>;
+using StringValueGetterRaw = StringValueGetterImpl<false>;
+
+// Value getters for `isIRI`, `isBlank`, and `isLiteral`.
+template <auto isSomethingFunction, auto prefix>
+struct IsSomethingValueGetter {
+  Id operator()(ValueId id, const EvaluationContext* context) const {
+    return Id::makeFromBool(id.getDatatype() == Datatype::VocabIndex &&
```

Comment:
Those currently will not work for the local vocab, but right move here is only to add a 
TODO<joka921, greenBene> because currently the local vocab is not quite correct with respect to Iris and Literals anyway.